RealmSwift/Tests/RealmCollectionTypeTests.swift

```diff
@@ -522,6 +522,66 @@ class RealmCollectionTypeTests: TestCase {
         XCTAssertTrue(collection.freeze().isFrozen)
     }
 
+    func testThaw() {
+        let frozen = collection.freeze()
+        XCTAssertTrue(frozen.isFrozen)
+
+        let frozenRealm = frozen.realm!
+        assertThrows(try! frozenRealm.write {}, reason: "Can't perform transactions on a frozen Realm")
+
+        let live = frozen.thaw()
+        XCTAssertFalse(live.isFrozen)
+
+        let liveRealm = live.realm!
+        try! liveRealm.write { liveRealm.delete(live) }
+        XCTAssertTrue(live.isEmpty)
+        XCTAssertFalse(frozen.isEmpty)
+    }
+    
+    func testThawFromDifferentThread() {
+        let frozen = collection.freeze()
+        XCTAssertTrue(frozen.isFrozen)
+
+        dispatchSyncNewThread {
+            let live = frozen.thaw()
+            XCTAssertFalse(live.isFrozen)
+
+            let liveRealm = live.realm!
+            try! liveRealm.write { liveRealm.delete(live) }
+            XCTAssertTrue(live.isEmpty)
+            XCTAssertFalse(frozen.isEmpty)
+        }
+    }
+    
+    func testThawPreviousVersion() {
```

Comment:
This is testing a thing that should be tested, but wasn't quite what I was referring to. An interesting scenario to verify is

1. Open realm on main thread and create object.
2. On background thread mutate object then freeze it
3. Thaw the frozen object on the main thread and verify that it has the value from before the write (as the main thread hasn't refreshed yet)

A similar scenario would be instead creating the object initially on the background thread so that when it's thawed on the main thread it won't exist. The most likely to be problematic case for that is a Results which filters a List which doesn't exist yet.